derive/src/root_query.rs

```diff
@@ -120,12 +120,42 @@ pub fn basic_query(name: &Ident, path: &TokenStream) -> TokenStream {
                             .await
                             .unwrap();
 
-                        let pages = paginator
-                            .num_pages()
+                        let sea_orm::ItemsAndPagesNumber { number_of_pages: pages, number_of_items: total_count } = paginator
+                            .num_items_and_pages()
                             .await
                             .unwrap();
 
-                        seaography::data_to_connection::<#path::Entity>(data, pagination.page != 1, pagination.page < pages, Some(pages), Some(pagination.page))
+                        seaography::data_to_connection::<#path::Entity>(
```

Comment:
Can you write a function that has the generic pagination query and use it in both Offset and Pagination cases ?
This will remove the code duplication.

If not its fine, I will take care of it in the new version :)

Reply:
Who is "you"?

Btw, what tools do you use to debug and write macro code and how do you test it?

Reply:
@estriv @karatakis  what is this pull request waiting for? Can I help somehow?

Reply:
@skopz356 From my side everything is fine.

Reply:
@karatakis Hi, what can I do to have this request merged?